pkg/cmd/dev/test.go

```diff
@@ -66,14 +66,14 @@ pkg/kv/kvserver:kvserver_test) instead.`,
     dev test pkg/spanconfig/... pkg/ccl/spanconfigccl/...
         Test multiple packages recursively
 
-    dev test --race --stress ...
-        Run a test under race and stress
+    dev test --race --count 250 ...
```

Comment:
`--count N` is a less helpful pattern than using `--stress` with or without a time limit ([internal discussion](https://cockroachlabs.slack.com/archives/C0HM2DZ34/p1683919924410619?thread_ts=1683844529.765849&cid=C0HM2DZ34)). This doesn't change even if `--count` starts using parallelism like is happening in this PR, or we nuke using the `stress` binary under `--stress` in some future PR. Could you surface some usage text that talks about `dev test --stress` specifically, ordered somewhere higher up on this list?

Reply:
> --count N is a less helpful pattern than using --stress with or without a time limit

Citation needed -- at best, this is very subjective. They do almost exactly the same thing, except `--count` has better Bazel support, will produce more comprehensible test logs, and requires fewer hacks under-the-hood.

In the interest of getting the change merged, I moved the `--stress` documentation line up above the `--count` documentation, despite it not being a better tool in actuality.

Reply:
w.r.t running a unit test until it fails, you have the approach mentioned [in this post](https://stackoverflow.com/questions/59857058/run-a-test-until-it-fails) which is sensible and doesn't require hacking in an unrelated binary to repeatedly run a test and merge the output artifacts as `stress` does.

Reply:
TIL about `--notest_keep_going`, thanks! Thanks for highlighting `--stress` in the documentation. I was speaking subjectively, it's hard to give citations for a CRDB-internal CLI's UX. That said, there are a lot more occurrences of people mentioning (and presumably using) `dev test ... --stress` in github comments over `dev test ... --count`:

- https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+dev+test+--stress
- https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+dev+test+--count (spot checking the results don't actually turn up any real uses, unlike the other search here)

Maybe it's because it wasn't parallelized until now, but this is pedantic. If we later rip out `--stress` entirely in favor of bazel powered --count, another subjective opinion is that the current "N runs so far, 0 failures, over duration" logging is useful if only to communicate to others how high/low a flake rate something has. Looks like folks other than me use it: 

- https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+%22runs+so+far%2C+0+failures%22+

Reply:
`stress` is very entrenched in CRL internally so it's not surprising to me that a lot of people use it. The argument is not that no one uses `stress` so it can be gotten rid of -- the argument is that we have Bazel-native tools that work as well or better, regardless of whether people are aware of them or not. Meanwhile, the purpose of `dev` is to surface these Bazel patterns that people are not necessarily aware of.

We will probably not just remove the `--stress` argument to `dev test` entirely and keep it (although it won't shell out to the `stress` binary, and will instead just use Bazel to run many tests). I'll send a PR out probably in the near future so you'll be able to see what I mean.